### PR TITLE
MINIFICPP-280 Move extension tests and test headers, lint extensions

### PR DESCRIPTION
**Pull request by calebj**
- move extension tests into their respective folders
- separate source and header files
- remove unnecessary or nonexisting include directories
- run linter on extension source files as part of linter target
- clean up extensions according to linter
- add ability to specify more than one include and source folder for linter

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [x] Does your PR title start with MINIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: extensions/http-curl/tests/integration/C2NullConfiguration.cpp [hunk 1]
```diff
@@ -29,26 +29,27 @@
...
 #include <iostream>
 #include <sstream>
-#include "HTTPClient.h"
-#include "InvokeHTTP.h"
-#include "../TestBase.h"
+#include "client/HTTPClient.h"
+#include "processors/InvokeHTTP.h"
 #include "utils/StringUtils.h"
 #include "core/Core.h"
-#include "../include/core/logging/Logger.h"
```
Comment (phrocker) [inline extensions/http-curl/tests/integration/C2NullConfiguration.cpp]: oh man. everytime I see these I get upset. Eclipse does this to me. I use eclipse cdt and intellij for java....but never can understand why eclipse does this on a fresh workspace.

File: thirdparty/google-styleguide/run_linter.sh [hunk 2]
```diff
@@ -14,16 +15,31 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-# ./run_linter <includedir> <srcdir>
-#!/bin/bash
-if [ "$(uname)" == "Darwin" ]; then
-SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+# ./run_linter <includedir1> <includedir2> ... <includedirN> -- <srcdir1> <srcdir2> ... <srcdirN>
```
Comment (phrocker) [inline thirdparty/google-styleguide/run_linter.sh]: I like this. I was hoping to make these linter changes following our upcoming release. I also like what you did with tests and was toying with that but didn't want to jump to that because I had a full plate with other features and cmake additions I really appreciate this PR as it helps us tremendously. I'll run through it. May not make it through to this release but we'll get this reviewed and merged as it's super helpful.

**Comment by phrocker** [comment]
@calebj my intent was to do some of these things following the release in separate tickets. Can you break these up? Thanks

**Comment by achristianson** [comment]
@calebj thank you for the clean up contribution. Looks good to me; concur with @phrocker's comments.

File: .travis.yml [hunk 3]
```diff
@@ -21,7 +21,7 @@ cache:
   timeout: 300
 
 env:
-   CMAKE_BUILD_OPTIONS="-DENABLE_PCAP=TRUE"
+   CMAKE_BUILD_OPTIONS="-DENABLE_PCAP=TRUE -DEMABLE_USB_CAMERA=1 -DENABLE_GPS=1"
```
Comment (phrocker) [inline .travis.yml]: Should this be ENABLE_USB_CAMERA ?

File: extensions/http-curl/client/HTTPClient.cpp [hunk 4]
```diff
@@ -87,26 +89,17 @@ HTTPClient::HTTPClient()
...
 HTTPClient::~HTTPClient() {
-  if (nullptr != headers_) {
-    curl_slist_free_all(headers_);
-    headers_ = nullptr;
-  }
-  if (http_session_ != nullptr) {
-    curl_easy_cleanup(http_session_);
-    http_session_ = nullptr;
-  }
+  forceClose();
```
Comment (phrocker) [inline extensions/http-curl/client/HTTPClient.cpp]: Why did you remove the destruction of http_session_ ? That means we can't reclaim resources and in some cases may result in a TCP socket in CLOSE_WAIT

**Comment by phrocker** [comment]
@calebj Obviously this PR is a bit dated in code but a positive step forward. Everytime I notice structural issues in code in extensions I think back to this PR. If I do the leg work to get this PR up to date with the current code base would you be able to take a look at my updates? I'll simply submit a PR against your branch, so you would still get the credit for the great work you did. Thanks!

**Comment by calebj** [comment]
@phrocker Sure thing. I'm not with the project that needed minifi anymore, but I'm still interested in seeing these changes implemented.

**Comment by arpadboda** [comment]
This PR looks abandoned and seems to require a lot of effort to rebase, @phrocker do you think it worth to be kept open?

**Comment by phrocker** [comment]
@arpadboda I started a branch as of my last comment -- but it's only a small part of my work, so it will take a bit of time. I don't think there is any harm in keeping it open
